Slicetools update breaks local tests

Tests are failing for me locally. Assuming that the problem was that I am using the latest hdf5 version, I tried using both the the stable version of `hdf5` and `hdf5-git` from the AUR but both fail to pass most tests:

```bash
$ pytest . --import-mode=importlib versioned_hdf5/tests/test_api.py::test_stage_version
============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.12.3, pytest-8.2.2, pluggy-1.5.0
project deps: h5py-3.11.0, ndindex-1.8
rootdir: /home/pdmurray/Desktop/workspace/versioned-hdf5
configfile: pyproject.toml
plugins: hypothesis-6.104.2, env-1.1.3
collected 1 item                                                                                                                                                                                                 

versioned_hdf5/tests/test_api.py F                                                                                                                                                                         [100%]

==================================================================================================== FAILURES ====================================================================================================
_______________________________________________________________________________________________ test_stage_version _______________________________________________________________________________________________

vfile = <VersionedHDF5File object "/tmp/pytest-of-pdmurray/pytest-7/test_stage_version0/file.hdf5" (mode r+)>

    def test_stage_version(vfile):
        """Test that versions can be staged and are the expected shape."""
        test_data = np.concatenate(
            (
                np.ones((2 * DEFAULT_CHUNK_SIZE,)),
                2 * np.ones((DEFAULT_CHUNK_SIZE,)),
                3 * np.ones((DEFAULT_CHUNK_SIZE,)),
            )
        )
    
        # Chunk size is intelligently selected based on first dataset written
        chunk_size = guess_chunk(test_data.shape, None, test_data.dtype.itemsize)[0]
        with vfile.stage_version("version1", "") as group:
            group["test_data"] = test_data
    
        version1 = vfile["version1"]
        assert version1.attrs["prev_version"] == "__first_version__"
        assert_equal(version1["test_data"], test_data)
    
        ds = vfile.f["/_version_data/test_data/raw_data"]
    
        # The dataset has 3 different arrays of size chunk_size, so should have
        # shape 3*chunk_size
        assert ds.shape == (3 * chunk_size,)
        assert_equal(ds[0 : 1 * chunk_size], 1.0)
        assert_equal(ds[1 * chunk_size : 2 * chunk_size], 2.0)
        assert_equal(ds[2 * chunk_size : 3 * chunk_size], 3.0)
    
        with vfile.stage_version("version2", "version1") as group:
>           group["test_data"][0] = 0.0

versioned_hdf5/tests/test_api.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
versioned_hdf5/wrappers.py:1395: in __setitem__
    self.dataset.__setitem__(index, value)
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
versioned_hdf5/wrappers.py:1025: in __setitem__
    if c not in self.id.data_dict:
versioned_hdf5/wrappers.py:1446: in data_dict
    self._data_dict = build_data_dict(dcpl, self.raw_data.name)
slicetools.pyx:124: in versioned_hdf5.slicetools.build_data_dict
    ???
slicetools.pyx:140: in versioned_hdf5.slicetools.build_data_dict
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: Could not get virtual filename

slicetools.pyx:164: ValueError
---------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------
HDF5-DIAG: Error detected in HDF5 (1.15.0):
  #000: /usr/src/debug/hdf5-git/hdf5/src/H5Pdcpl.c line 2397 in H5Pget_virtual_filename(): can't find object for ID
    major: Object ID
    minor: Unable to find ID information (already closed?)
  #001: /usr/src/debug/hdf5-git/hdf5/src/H5Pint.c line 4095 in H5P_object_verify(): property list is not a member of the class
    major: Property lists
    minor: Can't compare objects
  #002: /usr/src/debug/hdf5-git/hdf5/src/H5Pint.c line 4046 in H5P_isa_class(): not a property list
    major: Invalid arguments to routine
    minor: Inappropriate type
================================================================================================ warnings summary ================================================================================================
../../../.pyenv/versions/3.12.3/envs/vhdf/lib/python3.12/site-packages/_pytest/config/__init__.py:1448
  /home/pdmurray/.pyenv/versions/3.12.3/envs/vhdf/lib/python3.12/site-packages/_pytest/config/__init__.py:1448: PytestConfigWarning: Unknown config option: import_mode
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================ short test summary info =============================================================================================
FAILED versioned_hdf5/tests/test_api.py::test_stage_version - ValueError: Could not get virtual filename
========================================================================================== 1 failed, 1 warning in 0.51s ==========================================================================================
```

OS: Linux 6.6.36-1-lts x86_64

## hdf5 config

```bash
$ h5cc -showconfig
/usr/h5cc
dir is /usr
        SUMMARY OF THE HDF5 CONFIGURATION
        =================================

General Information:
-------------------
                   HDF5 Version: 1.15.0
                  Configured on: 2024-07-02
                  Configured by: Unix Makefiles
                    Host system: Linux-6.6.36-1-lts
              Uname information: Linux
                       Byte sex: little-endian
             Installation point: /usr

Compiling Options:
------------------
                     Build Mode: Release
              Debugging Symbols: OFF
                        Asserts: OFF
                      Profiling: OFF
             Optimization Level: OFF

Linking Options:
----------------
                      Libraries: 
  Statically Linked Executables: OFF
                        LDFLAGS: -Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now          -Wl,-z,pack-relative-relocs -flto=auto
                     H5_LDFLAGS: 
                     AM_LDFLAGS: 
                Extra libraries: m;dl
                       Archiver: /usr/bin/ar
                       AR_FLAGS:
                         Ranlib: /usr/bin/ranlib

Languages:
----------
                              C: YES
                     C Compiler: /usr/bin/cc 14.1.1
                       CPPFLAGS: 
                    H5_CPPFLAGS: 
                    AM_CPPFLAGS: 
                         CFLAGS:   -std=c99 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/home/pdmurray/.cache/paru/clone/hdf5-git/src=/usr/src/debug/hdf5-git -flto=auto -fstdarg-opt -fdiagnostics-urls=never -fno-diagnostics-color -fmessage-length=0
                      H5_CFLAGS: -Wall;-Wcast-qual;-Wconversion;-Wextra;-Wfloat-equal;-Wformat=2;-Winit-self;-Winvalid-pch;-Wmissing-include-dirs;-Wshadow;-Wundef;-Wwrite-strings;-pedantic;-Wno-c++-compat;-Wbad-function-cast;-Wcast-align;-Wformat;-Wimplicit-function-declaration;-Wint-to-pointer-cast;-Wmissing-declarations;-Wmissing-prototypes;-Wnested-externs;-Wold-style-definition;-Wpacked;-Wpointer-sign;-Wpointer-to-int-cast;-Wredundant-decls;-Wstrict-prototypes;-Wswitch;-Wunused-but-set-variable;-Wunused-variable;-Wunused-function;-Wunused-parameter;-finline-functions;-Wno-aggregate-return;-Wno-inline;-Wno-missing-format-attribute;-Wno-missing-noreturn;-Wno-overlength-strings;-Wlarger-than=2560;-Wlogical-op;-Wframe-larger-than=16384;-Wpacked-bitfield-compat;-Wsync-nand;-Wno-unsuffixed-float-constants;-Wdouble-promotion;-Wtrampolines;-Wstack-usage=8192;-Wmaybe-uninitialized;-Wno-jump-misses-init;-Wstrict-overflow=2;-Wno-suggest-attribute=const;-Wno-suggest-attribute=noreturn;-Wno-suggest-attribute=pure;-Wno-suggest-attribute=format;-Wdate-time;-Warray-bounds=2;-Wc99-c11-compat;-Wincompatible-pointer-types;-Wint-conversion;-Wshadow;-Wduplicated-cond;-Whsa;-Wnormalized;-Wnull-dereference;-Wunused-const-variable;-Walloca;-Walloc-zero;-Wduplicated-branches;-Wformat-overflow=2;-Wformat-truncation=1;-Wrestrict;-Wattribute-alias;-Wshift-overflow=2;-Wcast-function-type;-Wmaybe-uninitialized;-Wcast-align=strict;-Wno-suggest-attribute=cold;-Wno-suggest-attribute=malloc;-Wattribute-alias=2;-Wmissing-profile;-Wc11-c2x-compat
                      AM_CFLAGS: 
               Shared C Library: YES
               Static C Library: YES

                        Fortran: OFF
               Fortran Compiler:  
                  Fortran Flags: 
               H5 Fortran Flags: 
               AM Fortran Flags: 
         Shared Fortran Library: YES
         Static Fortran Library: YES
               Module Directory: /home/pdmurray/.cache/paru/clone/hdf5-git/src/hdf5/_build/mod

                            C++: OFF
                   C++ Compiler:  
                      C++ Flags: 
                   H5 C++ Flags: 
                   AM C++ Flags: 
             Shared C++ Library: YES
             Static C++ Library: YES

                           JAVA: OFF
                  JAVA Compiler:  

Features:
---------
                     Parallel HDF5: OFF
  Parallel Filtered Dataset Writes: 
                Large Parallel I/O: 
                High-level library: ON
Dimension scales w/ new references: 
                  Build HDF5 Tests: ON
                  Build HDF5 Tools: ON
                   Build GIF Tools: OFF
                      Threadsafety: OFF
               Default API mapping: v116
    With deprecated public symbols: ON
            I/O filters (external):  DEFLATE
                  _Float16 support: ON
                     Map (H5M) API: OFF
                        Direct VFD: OFF
                        Mirror VFD: OFF
                     Subfiling VFD: OFF
                (Read-Only) S3 VFD: OFF
              (Read-Only) HDFS VFD: OFF
    Packages w/ extra debug output: 
                       API Tracing: OFF
              Using memory checker: OFF
                  Use file locking: best-effort
         Strict File Format Checks: OFF
      Optimization Instrumentation: 
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -95,7 +95,7 @@ ignore = [
 "test_*.py" = ["ANN001"]
 
 [project.optional-dependencies]
-dev = ["pre-commit>=3.6.0"]
+dev = ["pre-commit>=3.6.0", 'cython', 'meson-python', 'setuptools-scm']
 test = ["pytest", "pytest-env", "hypothesis"]
 doc = ["sphinx", "sphinx-multiversion", "myst-parser"]
 

--- a/versioned_hdf5/slicetools.pyx
+++ b/versioned_hdf5/slicetools.pyx
@@ -1,7 +1,7 @@
 from functools import lru_cache
 
 from ndindex import Slice, Tuple
-from h5py import h5d, h5s
+from h5py import h5d, h5s, h5i
 from h5py._hl.base import phil
 
 import cython
@@ -12,11 +12,15 @@ from libc.string cimport strlen, strncmp
 from libcpp.vector cimport vector
 
 cdef extern from "hdf5.h":
-    # HDF5 types
+    # HDF5 types copied from h5py/api_types_hdf5.pxd
     ctypedef long int hid_t
+    ctypedef int hbool_t
     ctypedef int herr_t
     ctypedef int htri_t
     ctypedef long long hsize_t
+    ctypedef signed long long hssize_t
+    ctypedef signed long long haddr_t
+    ctypedef long int off_t
 
     # virtual Dataset functions
     cdef ssize_t H5Pget_virtual_dsetname(hid_t dcpl_id, size_t index, char *name, size_t size)
@@ -74,7 +78,7 @@ def hyperslab_to_slice(start, stride, count, block):
 cdef _spaceid_to_slice(space_id: hid_t):
     """
     Helper function to read the data for `space_id` selection and
-    convert it to a Tuple of slices. 
+    convert it to a Tuple of slices.
     """
     sel_type: H5S_sel_type = H5Sget_select_type(space_id)
 
@@ -124,13 +128,13 @@ cdef _spaceid_to_slice(space_id: hid_t):
 cpdef build_data_dict(dcpl, raw_data_name: str):
     """
     Function to build the "data_dict" of a versioned virtual dataset.
-    
+
     All virtual datasets created by versioned-hdf5 should have chunks in
     exactly one raw dataset `raw_data_name` in the same file. This function will
-    check that this is the case and return a dictionary mapping the `Tuple` of 
+    check that this is the case and return a dictionary mapping the `Tuple` of
     the chunk in the virtual dataset to a `Slice` in the raw dataset.
-    
-    :param dcpl: the dataset creation property list of the versioned dataset  
+
+    :param dcpl: the dataset creation property list of the versioned dataset
     :param raw_data_name: the name of the corresponding raw dataset
     :return: a dictionary mapping the `Tuple` of the virtual dataset chunk
         to a `Slice` in the raw dataset.

